src/test/java/org/testng/internal/dynamicgraph/EdgeWeightTestSample1.java

```diff
@@ -0,0 +1,27 @@
+package org.testng.internal.dynamicgraph;
+
+import java.util.List;
+
+import org.testng.annotations.Test;
+import org.testng.collections.Lists;
+
+public class EdgeWeightTestSample1 {
+
+    public static List<String> testCaseExecutionOrder = Lists.newArrayList();
+
+    @Test(groups = { "g1" }, priority = 2)
+    public void t1() {
+        testCaseExecutionOrder.add(new Object() {}.getClass().getEnclosingMethod().getName());
```

Comment:
I think `testCaseExecutionOrder.add("t1)` is more readable and cost less characters ;)